examples/compatibility-6-wind-router-rules.yaml

```diff
@@ -0,0 +1,31 @@
+version: v1alpha1
+compatibilities:
+- description: "required kernel configuration"
+  rules:
+  - name: "kernel configuration"
+    matchFeatures:
+    - feature: kernel.config
+      matchExpressions:
+        CGROUP_BPF: {op: In, value: ["y"]}
+        XFRM_INTERFACE: {op: In, value: ["y"]}
+        NF_TABLES: {op: In, value: ["y"]}
+        NETFILTER_XT_TARGET_NOTRACK: {op: In, value: ["y"]}
+        NET_ACT_BPF: {op: In, value: ["y"]}
+        MPLS_ROUTING: {op: In, value: ["y"]}
+        MPLS_IPTUNNEL: {op: In, value: ["y"]}
+        PPPOE: {op: In, value: ["y"]}
+    - feature: kernel.loadedmodule
+      matchExpressions:
+        br_netfilter: {op: Exists}
+        ebtables: {op: Exists}
+        ifb: {op: Exists}
+        ip6_tables: {op: Exists}
+        ip_tables: {op: Exists}
+        mpls_iptunnel: {op: Exists}
+        mpls_router: {op: Exists}
+        nf_conntrack: {op: Exists}
+        overlay: {op: Exists}
+        ppp_generic: {op: Exists}
+        vfio-pci: {op: Exists}
+        vhost-net: {op: Exists}
+        vrf: {op: Exists}
```

Comment:
how about add another example, for example:
```
version: v1alpha1
compatibilities:
- description: "intel cpu and ubuntu OS"
  rules:
    - name: "ubuntu"
      matchFeatures:
        - feature: cpu.model
          matchExpressions:
            vendor_id: { op: In, value: [ "Intel" ] }
        - feature: kernel.config
          matchExpressions:
            CGROUP_BPF: { op: In, value: [ "y" ] }
            XFRM_INTERFACE: { op: In, value: [ "y" ] }
            NF_TABLES: { op: In, value: [ "y" ] }
            NETFILTER_XT_TARGET_NOTRACK: { op: In, value: [ "y" ] }
        - feature: kernel.loadedmodule
          matchExpressions:
            br_netfilter: { op: Exists }
            ebtables: { op: Exists }
            ifb: { op: Exists }
            ip_tables: { op: Exists }
            nf_conntrack: { op: Exists }
            overlay: { op: Exists }
            ppp_generic: { op: Exists }
            vfio-pci: { op: Exists }
            vhost-net: { op: Exists }
            vrf: { op: Exists }
        - feature: system.osrelease
          matchExpressions:
            ID: { op: In, value: [ "ubuntu" ] }
            VERSION_ID: { op: In, value: [ "20.04", "22.04" ] }
    - name: "nic"
      matchAny:
        - matchFeatures:
          - feature: pci.device
            matchExpressions:
              vendor: {op: In, value: ["0eee"]}
              class: {op: In, value: ["0200"]}
        - matchFeatures:
          - feature: pci.device
            matchExpressions:
              vendor: {op: In, value: ["0fff"]}
              class: {op: In, value: ["0200"]}
- description: "Amd cpu and RHEL"
  rules:
    - name: "RHEL"
      matchFeatures:
        - feature: cpu.model
          matchExpressions:
            vendor_id: { op: In, value: [ "Amd" ] }
        - feature: kernel.config
          matchExpressions:
            CGROUP_BPF: { op: In, value: [ "y" ] }
            MPLS_ROUTING: { op: In, value: [ "y" ] }
            MPLS_IPTUNNEL: { op: In, value: [ "y" ] }
            PPPOE: { op: In, value: [ "y" ] }
        - feature: kernel.loadedmodule
          matchExpressions:
            ip6_tables: { op: Exists }
            ip_tables: { op: Exists }
            mpls_iptunnel: { op: Exists }
            mpls_router: { op: Exists }
            vfio-pci: { op: Exists }
            vhost-net: { op: Exists }
            vrf: { op: Exists }
        - feature: system.osrelease
          matchExpressions:
            ID: { op: In, value: [ "rhel" ] }
            VERSION_ID: { op: In, value: [ "8" ] }
```
in which illustrates other way to organize the compatibilities, and show how matchAny could be used in the rule

Reply:
What about placing something more simple (as introduced in the proposal)?
```yaml
version: v1alpha1
compatibilities:
- tag: "prefered"
  weight: 10
  description: "Prefered node configuration"
  rules:
  - name: "kernel and cpu"
    matchFeatures:
    - feature: kernel.loadedmodule
      matchExpressions:
        vfio-pci: {op: Exists}
    - feature: cpu.model
      matchExpressions:
        vendor_id: {op: In, value: ["Intel", "Amd"]}
- tag: "fallback"
  weight: 1
  description: "Minimal required configuration"
  rules:
  - name: "cpu"
    matchFeatures:
    - feature: cpu.model
      matchExpressions:
        vendor_id: {op: In, value: ["Intel", "Amd"]}
```

The example is easy to understand and run.

Reply:
o, it's not about simplification, it's to include "matchAny" in the example

Reply:
I will incldue it.

Reply:
I've also added the backreference and matchAny features to the example.